synapse/storage/events_bg_updates.py

```diff
@@ -397,3 +401,60 @@ def _drop_table_txn(txn):
             )
 
         return num_handled
+
+    @defer.inlineCallbacks
+    def _redactions_received_ts(self, progress, batch_size):
+        """Handles filling out the `received_ts` column in redactions.
+        """
+        last_event_id = progress.get("last_event_id", "")
+
+        def _redactions_received_ts_txn(txn):
+            # Fetch the set of event IDs that we want to update
+            sql = """
+                SELECT event_id FROM redactions
+                WHERE event_id > ?
+                ORDER BY event_id ASC
+                LIMIT ?
```

Comment:
You might want to also exclude the redactions for which `received_ts` is not `NULL`, so that you don't process redactions that we received between two calls to this function.

Reply:
Yeah, could do! I don't think it makes much of a difference (it'll do the right thing anyway), so I'm probably just going to leave it as is.